resources/table--catalog.sql

```diff
@@ -0,0 +1,22 @@
+create table if not exists catalog (
+    source enum('wowinterface', 'curseforge'),
+    source_id int,
+
+    label varchar(255),
+    name varchar(150), -- longest in catalog at time of writing is 49
+    alt_name varchar(150),
+    description varchar(255),
+    uri varchar(255),
+    download_count int,
+    -- created_date timestamp with time zone, -- curseforge only and unused
+    -- updated_date timestamp with time zone, -- nobody can seem to do dates without *fucking them up* somehow
+    updated_date varchar(24), -- we have dates with micro second precision 
+
+    retail_track boolean,
+    vanilla_track boolean,
+    
+    primary key(source_id, source)
```

Comment:
update todo: compare catalog loading speeds when addon category associations are made as we go along as opposed to in three blocks (categories, addons, category-addons). We might save some time and code